packages/react-icon-provider/src/IconProvider.tsx

```diff
@@ -0,0 +1,23 @@
+import { IIconSubset } from '@fluentui/style-utilities';
+import * as React from 'react';
+import { IconProviderProps } from './IconProvider.types';
+
+/**
+ * Context for providing the window.
+ */
+export const IconContext = React.createContext<IconProviderProps>({
+  icons: { icons: {} },
+});
+
+/**
+ * Hook to access the Icon object. This can be overridden contextually using the `IconProvider`.
+ */
+export const useIcon = (): IIconSubset | undefined => React.useContext(IconContext).icons;
```

Comment:
Should this be called `useIconSubset` or `useIcons`? `useIcon` (singular) seems weird since it's actually a set of things. Though the downside is that would make the naming slightly mismatched from IconProvider...